ui/src/components/navigation/Sidebar.tsx

```diff
@@ -26,44 +31,171 @@ const SideBar: FC<SidebarProps> = ({ isOpen }): ReactElement => {
   const navigate = useNavigate();
   const [selectedItem, setSelectedItem] = useState<string | null>(null);
   const isDesktop = useMediaQuery('(min-width: 769px)');
-  const menuItems = [
+
+  const [allSongs, setAllSongs] = useState<SongSearchModalProps[]>([]);
+  const [allSetlists, setAllSetlists] = useState<SetlistSearchModalProps[]>([]);
+
+  const getAllSongs = useCallback(async () => {
+    try {
+      const { data, status } = await axios.get('/api/songs/get');
+      if (status === 200) {
+        setAllSongs(data);
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
+  useEffect(() => {
+    getAllSongs();
+  }, [getAllSongs]);
+
+  const getAllSetlists = useCallback(async () => {
+    try {
+      const setlistRes = await axios.get('/api/setlists/get');
+      if (setlistRes.status === 200) {
+        setAllSetlists(setlistRes.data);
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  }, []);
+
+  useEffect(() => {
+    getAllSetlists();
+  }, [getAllSetlists]);
```

Comment:
I think its better if you move this logic to the search modal, or only trigger when the search modal is opened. If not it will call all the songs and setlists on every page, since we have the sidebar on every page.
It will also call the songs API on the songs page, causing multiple redundant API calls.